examples/event-loop/signals.php

```diff
@@ -9,7 +9,20 @@
 
 Loop::onSignal(SIGINT, function () {
     print "Caught SIGINT, exiting..." . PHP_EOL;
+    
+    // Check for a Uv driver
+    if (Loop::get() instanceof Amp\Loop\UvDriver) {
+
+        // Stop the loop
+        Loop::stop();
+
+        // Cannot exit out of a UvDriver loop here, can only stop the loop
+        return;
+    }
+    
```

Comment:
What happens if you cancel the signal watcher before existing? Did you look where exactly it hangs in probably an infinite loop causing the 100% CPU usage?

```suggestion
Loop::onSignal(SIGINT, function ($watcherId) {
    print "Caught SIGINT, exiting..." . PHP_EOL;
    
    Loop::cancel($watcherId);
    
```

Reply:
Not sure where my comment ended up, so sorry if this is a double tap.

Just tried this code and it still seems to get stuck in a loop. I tried debugging it for a couple hours, but I didn't get anywhere. Xdebug seemed to dump the profiler data before it got stuck in the loop, which made me think it was a `Uv` lib problem. (trying without xdebug loaded results in the same problem)